src/movies.js

```diff
@@ -1,25 +1,96 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+
+function getAllDirectors(moviesArray){
+    const directorsArray = moviesArray.map(element => element.director, 0);
+    return directorsArray
+}
+
+
+function cleanAllDirectors(moviesArray){
+    let newArr = [];
+    for (let i = 0; i<moviesArray.length; i++){
+        if (newArr !== moviesArray[i].director){
+            newArr.push(moviesArray[i])
+        }
+    }
+}
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray){
+    if (moviesArray.length === 0){
+        return 0;
+    }
+    const spielbergOnly = moviesArray.filter(element => element.director === 'Steven Spielberg' && element.genre.includes('Drama'));
+    return spielbergOnly.length;
+}
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+    if (moviesArray.length === 0){
+        return 0;
+    }
+    const totalScores = moviesArray.reduce((sum, element) => sum + element.score, 0);
```

Comment:
Here in order to get the last iteration correct you need to create an if statement to check  if the score property its coming 

```suggestion
    const totalScores =moviesArray.reduce((total, current) => {
        if (current.score) {
            return total + current.score
        } else {
            return total
        }
    }, 0)
```